.github/workflows/tests/smoke_test_auth_cherry_picker.sh

```diff
@@ -0,0 +1,40 @@
+# Replace our build support files with stub ones
+cat << EOF > build-support/bin/cherry_pick/make_pr.sh
+#!/usr/bin/env bash
+echo "I would've made a PR!"
+# We exit 1 to test we still call the finish job
+exit 1
+EOF
+chmod +x build-support/bin/cherry_pick/make_pr.sh
+
+cat << EOF > build-support/bin/cherry_pick/helper.js
+class CherryPickHelper {
+    constructor({ octokit, context, core }) {}
+    async get_prereqs() {
+        return {
+            pr_num: 12345,
+            merge_commit: "ABCDEF12345",
+            milestones: ["2.16.x", "2.17.x"],
+        };
+  }
+
+    async cherry_pick_finished(merge_commit_sha, matrix_info) {
+        console.log(\`We finished: \${merge_commit_sha} \${matrix_info}\`);
+    }
+};
+
+module.exports = CherryPickHelper;
+EOF
+
+
+OUTPUT=$(./act workflow_dispatch -W .github/workflows/auto-cherry-picker.yaml --input PR_number=17295 --env GITHUB_REPOSITORY=pantsbuild/pants)
```

Comment:
I think shells do slightly weird things with process capture like this and exit codes. I think a typical alternative is to instead capture to files:

```suggestion
output_file=__output.txt
./act workflow_dispatch -W .github/workflows/auto-cherry-picker.yaml --input PR_number=17295 --env GITHUB_REPOSITORY=pantsbuild/pants > "$output_file"
```

The assertions can then become something like

```shell
if ! grep -F "make_pr.sh 12345 2.16.x" "$output_file"; then ... fi
if ! grep -F "make_pr.sh 12345 2.17.x" "$output_file"; then ... fi
if ! grep -F "cherry_picked_finished: ABCDEF12345 ..." "$output_file"; then ... fi
```

Doing this as follow-up is fine by me.